### PR TITLE
WIP: Thu 13 06 24 including js

### PR DESCRIPTION
**Pull request by psweeting1**
- Node script to copy GOVUK frontend JS into public directory.
- Set up project to use GOVUK template.njk with new njk file structure. 
- Index.njk file created
- Forms page created
- Response compression set up.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,12 @@
...
\ No newline at end of file
+.idea/
+
+# Visual Studios
+.trunk/
+
+# MOJ Express Skeleton
+public/images/*
+public/javascripts/*
+public/stylesheets/*
```
Comment (imtiazAhmd) [inline .gitignore:13]: Do we need anything from public folder ? May be we can ignore anything from public/* ?

File: package.json [hunk 2]
```diff
@@ -4,8 +4,9 @@
   "private": true,
   "scripts": {
     "start": "node ./bin/www",
-    "dev": "nodemon ./bin/www & npm run scss",
-    "scss": "sass --watch src/scss/main.scss public/stylesheets/style.css"
+    "dev": "nodemon ./bin/www & npm run scss & npm run js",
+    "scss": "sass --watch src/scss/main.scss public/stylesheets/govuk-frontend.min.css --style=compressed",
+    "js": "cp -R node_modules/govuk-frontend/dist/govuk/govuk-frontend.min.js* public/javascripts"
```
Comment (imtiazAhmd) [inline package.json:9]: May be we can have a compile script:
`"compile": "sass src/scss/main.scss public/stylesheets/style.css && cp -R node_modules/govuk-frontend/dist/govuk/govuk-frontend.min.js* public/javascripts"`

File: package.json [hunk 3]
```diff
@@ -4,8 +4,9 @@
   "private": true,
   "scripts": {
     "start": "node ./bin/www",
-    "dev": "nodemon ./bin/www & npm run scss",
-    "scss": "sass --watch src/scss/main.scss public/stylesheets/style.css"
+    "dev": "nodemon ./bin/www & npm run scss & npm run js",
+    "scss": "sass --watch src/scss/main.scss public/stylesheets/govuk-frontend.min.css --style=compressed",
```
Comment (imtiazAhmd) [inline package.json:8]: We may not name our compiled css as 'govuk-frontend'. Project specific scss will be in main.scss as well.

**Review by imtiazAhmd — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (psweeting1) [reply .gitignore:13]: Well, I don't think we generate the public folder when we need to use it, so I just excluded the content that we create. However we could tweak this so it does create a public folder when we do a `npm run`

File: package.json (hunk 3 above)
Comment (psweeting1) [reply package.json:8]: Agreed, maybe the generic (application.min.css etc) is better than saying GOVUK.

File: package.json (hunk 2 above)
Comment (psweeting1) [reply package.json:9]: OHHH I like this, adding that one in.

Comment (psweeting1) [reply package.json:9]: Actually I tried that and it's massive! I think separating and isolating them is fine and allows use to catch any script that fails.

Maintenance wise this also makes sense than having one long script.

**Review by imtiazAhmd — APPROVED** [review]
Lets merge it , so it unblocks other ticket implementation.

**Pull request merged**